plugins/webp-uploads/settings.php

```diff
@@ -113,13 +118,29 @@ function webp_uploads_generate_avif_webp_setting_callback(): void {
 
 	$selected       = webp_uploads_get_image_output_format();
 	$avif_supported = webp_uploads_mime_type_supported( 'image/avif' );
-	// Ensure WebP selected if AVIF is not supported.
-	if ( ! $avif_supported ) {
+	$webp_supported = webp_uploads_mime_type_supported( 'image/webp' );
+
+	// If neither format is support, the entire field is not shown.
+	if ( ! $avif_supported && ! $webp_supported ) {
+		?>
+		<br />
+		<div class="notice notice-warning inline">
+			<p><b><?php esc_html_e( 'Modern Image support is not available.', 'webp-uploads' ); ?></b></p>
+			<p><?php esc_html_e( 'WebP or AVIF support can only be enabled by your hosting provider, so contact them for more information.', 'webp-uploads' ); ?></p>
+		</div>
+		<?php
+		return;
+	}
+
+	// If only one of the two formats is supported, the dropdown defaults to that type and the other type is disabled.
+	if ( $avif_supported ) {
+		$selected = 'avif';
+	} else {
```

Comment:
I guess this is what I should have suggested instead.

```suggestion
	if ( ! $avif_supported && 'avif' === $selected ) {
		$selected = 'webp';
```